### PR TITLE
Remove similarity heuristic

### LINKED ISSUE #136: similarity_heuristic is bad
The code smells and it has a ton of false positives and false negatives. I couldn't even describe to you what the function does without specifying the names of the _multiple_ algorithms involved as well as the order and limits with which it processes match attempts. It would better off returning `random.choice([0, 3])`. This isn't to say that there's a better algorithm that can be used or that the code ought to be cleaned up, though those are probably both true statements, but the fundamental problem with requiring a similar username and real name is that not only is that similarity subjective, but nobody even knows what it means. That makes this not just a code smell, but a UX smell.

- If you want to be real draconian about having usernames be real names, generate the usernames programmatically and don't let the user change it without contacting humans (to deal with the edge case that your programatic solution makes their name sound offensive or something).
- Probably the best thing to do is remove the check and allow some freedom.

### PR DESCRIPTION
**Pull request by bjb568**
Fixes #136.

### CODE REVIEW

File: tests/account/creation_test.py [hunk 1]
```diff
@@ -212,75 +211,6 @@ def test_ensure_web_dir_rename(self):
...
-        """Test some obviously good and bad usernames."""
-        try:
-            validate_username(username, realname)
-        except ValidationWarning as ex:
-            if success:
-                pytest.fail(
-                    'Received unexpected error: {error}'.format(error=ex),
-                )
-
-    @pytest.mark.parametrize('username', [
```
Comment (jvperrin) [inline tests/account/creation_test.py:239]: This and everything below is are not related to the similarity heuristic, so they shouldn't be removed.

Comment (jvperrin) [reply tests/account/creation_test.py:239]: Well ok they reference it, but they're testing some other functionality as well

File: tests/account/creation_test.py [hunk 2]
```diff
@@ -212,75 +211,6 @@ def test_ensure_web_dir_rename(self):
...
-            validate_username(username, username)
-
-    def test_error_user_exists(self):
-        """Ensure that we raise an error if the username already exists."""
-        with pytest.raises(ValidationError):
-            validate_username('ckuehl', 'Chris Kuehl')
-
-    @mock.patch('ocflib.account.validators.validate_username')
-    @mock.patch('ocflib.account.search.user_exists', return_value=False)
-    def test_long_names(self, _, __):
```
Comment (jvperrin) [inline tests/account/creation_test.py:271]: This one is still fine to remove though, the permutation issue was only with the similarity heuristic.

**Comment by bjb568** [comment]
This PR is pending BoD approval.

**Comment by bjb568** [comment]
Approved by BoD.

File: tests/account/creation_test.py [hunk 3]
```diff
@@ -474,25 +436,6 @@ def test_valid_request(
             session,
         ) == ([], [])
 
-    @pytest.mark.parametrize('attrs', [
```
Comment (ja5087) [inline tests/account/creation_test.py:477]: This unit tests a lot of other functionality which is not a good sign but probably shouldn't be removed.

**Review by ja5087 — CHANGES_REQUESTED** [review]
otherwise lgtm pending that one test

**Review by ja5087 — APPROVED** [review]
Actually, looking at that test again I am not really sure what it's meant to warn against and it shouldn't block this PR.

**Pull request merged**